### PR TITLE
On scroll callback and set_position for scrollbars

### PR DESCRIPTION
**Pull request by johannphilippe**
* `std::function<void(point p)> on_scroll` lambda added to scroller_base (in `scroll`, and in `reposition`)
* `set_position(point p)` method added to `scroller_base` allowing to reposition the inside scroller (without calling `on_scroll` callback, since I use it to synchronize several scrollers, it would be infinite call).

### CODE REVIEW

File: examples/sync_scrollbars/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+set (CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${ELEMENTS_ROOT}/cmake")
+
+# If we are building outside the project, you need to set ELEMENTS_ROOT:
+if (NOT ELEMENTS_BUILD_EXAMPLES)
+   include(ElementsConfigCommon)
+   set(ELEMENTS_BUILD_EXAMPLES OFF)
+   add_subdirectory(${ELEMENTS_ROOT} elements)
+endif()
+
+set(ELEMENTS_APP_PROJECT "EmptyStarter")
```
Comment (djowel) [inline examples/sync_scrollbars/CMakeLists.txt:26]: EmptyStarter? Shouldn't this be a SyncScrollbars?

File: examples/sync_scrollbars/main.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using namespace cycfi::artist;
+
+// Main window background color
+auto constexpr bkd_color = rgba(35, 35, 37, 255);
+auto background = box(bkd_color);
+
+
+int main(int argc, char* argv[])
+{
+   app _app(argc, argv, "Empty Starter", "com.cycfi.empty-starter");
```
Comment (djowel) [inline examples/sync_scrollbars/main.cpp:18]: Probably bad name?

File: lib/src/element/port.cpp [hunk 3]
```diff
@@ -19,7 +21,7 @@ namespace cycfi { namespace elements
    void port_base::draw(context const& ctx)
    {
       auto state = ctx.canvas.new_state();
-      ctx.canvas.rect(ctx.bounds);
+      ctx.canvas.add_rect(ctx.bounds);
```
Comment (djowel) [inline lib/src/element/port.cpp:24]: These are artist APIs, right?

**Comment by djowel** [comment]
As expected, the build is breaking. It seems you targeted the artist library?

**Comment by johannphilippe** [comment]
I made a mistake actually. Since I experiment with artist, I just copied the wrong port and example. It should be fixed in last commit

File: examples/sync_scrollbars/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+if (NOT ELEMENTS_BUILD_EXAMPLES)
+   include(ElementsConfigCommon)
+   set(ELEMENTS_BUILD_EXAMPLES OFF)
+   add_subdirectory(${ELEMENTS_ROOT} elements)
+endif()
+
+set(ELEMENTS_APP_PROJECT "SyncScrollbars")
+set(ELEMENTS_APP_TITLE "Sync Scrollbars")
+set(ELEMENTS_APP_COPYRIGHT "Copyright (c) 2016-2020 Joel de Guzman")
+set(ELEMENTS_APP_ID "com.cycfi.empty-starter")
```
Comment (djowel) [inline examples/sync_scrollbars/CMakeLists.txt:27]: Perhaps this should be "com.cycfi.sync-scrollbars" ?

**Review by djowel — CHANGES_REQUESTED** [review]

**Pull request merged**